agglayer/types/types.go

```diff
@@ -123,6 +123,119 @@ const (
 	LeafTypeMessage
 )
 
+type AggchainData interface {
+	MarshalJSON() ([]byte, error)
+	UnmarshalJSON(data []byte) error
+	IsAggchainData() bool
+}
```

Comment:
WDYT to leverage to already existing interfaces in Golang?

```suggestion
type AggchainData interface {
	json.Marshaler
	json.Unmarshaler
	IsAggchainData() bool
}
```

Reply:
Btw what is the purpose of `IsAggchainData` function? As far I can tell, it always returns `true`...

Reply:
Fix: https://github.com/agglayer/aggkit/pull/221/commits/3fcd74ae75e58b72998eb78d7fc454140a82e721